apps/els_core/src/els_config_runtime.erl

```diff
@@ -31,6 +35,17 @@ get_node_name() ->
     Value = maps:get("node_name", els_config:get(runtime), default_node_name()),
     els_utils:compose_node_name(Value, get_name_type()).
 
+-spec get_hostname() -> string().
+get_hostname() ->
+    case els_config:get(runtime) of
+        undefined -> default_hostname();
+        Runtime -> maps:get("hostname", Runtime, default_hostname())
+    end.
```

Comment:
We want the `get_node_name` function to be able to use `get_hostname` but when the `els_config` module initializes this value is not ready yet, so we need to check for `undefined` here and allow it pass otherwise.